File: go/693.go
package main

func hasAlternatingBits(n int) bool {
	flag := n & 1
	n >>= 1

	for n > 0 {
		last := n & 1
		if last == flag {
			return false
		}

		flag = last
		n >>= 1
	}

	return true
}

File: zig/693.zig
const std = @import("std");

fn solution(n: usize) bool {
    var temp: usize = n;
    var flag: usize = n & 1;
    temp >>= 1;

    while (temp > 0) {
        const last: usize = temp & 1;
        if (last == flag) {
            return false;
        }

        flag = last;
        temp >>= 1;
    }

    return true;
}

const expect = std.testing.expect;

test "test case #1" {
    const input = 5;
    const expected = true;

    try expect(solution(input) == expected);
}

test "test case #2" {
    const input = 7;
    const expected = false;

    try expect(solution(input) == expected);
}

test "test case #3" {
    const input = 11;
    const expected = false;

    try expect(solution(input) == expected);
}
